internal/nodes/avi_model_nodes.go

```diff
@@ -721,80 +721,74 @@ func (o *AviVsNode) ReplaceSniSSLRefInSNINode(newSslNode *AviTLSKeyCertNode, key
 	o.SSLKeyCertRefs = append(o.SSLKeyCertRefs, newSslNode)
 }
 
-func (o *AviVsNode) AddFQDNAliasesToHTTPPolicy(host string, hosts []string, key string) {
-
-	var hppMap *AviHostPathPortPoolPG
-	var redirectPorts *AviRedirectPort
-	// Find the hppMap and redirectPorts that matches the host
-	for _, policy := range o.HttpPolicyRefs {
-		for j := range policy.HppMap {
-			if utils.HasElem(policy.HppMap[j].Host, host) {
-				hppMap = &policy.HppMap[j]
-				break
-			}
-		}
-		for j := range policy.RedirectPorts {
-			if utils.HasElem(policy.RedirectPorts[j].Hosts, host) {
-				redirectPorts = &policy.RedirectPorts[j]
-				break
+func (o *AviVsNode) AddFQDNAliasesToHTTPPolicy(hosts []string, key string) {
+
+	// Update the hosts in the hppMap of child VS
+	if o.IsSNIChild || o.Dedicated {
+		for _, policy := range o.HttpPolicyRefs {
+			for j := range policy.HppMap {
+				policy.HppMap[j].Host = make([]string, len(hosts))
+				copy(policy.HppMap[j].Host, hosts)
 			}
+			policy.AviMarkers.Host = make([]string, len(hosts))
+			copy(policy.AviMarkers.Host, hosts)
 		}
-		if utils.HasElem(policy.AviMarkers.Host, host) {
-			for _, host := range hosts {
-				if !utils.HasElem(policy.AviMarkers.Host, host) {
-					policy.AviMarkers.Host = append(policy.AviMarkers.Host, host)
-				}
-			}
+		if o.IsSNIChild {
+			utils.AviLog.Debugf("key: %s, msg: Added hosts %v to HTTP policy for child VS %s", key, hosts, o.Name)
+			return
 		}
 	}
 
-	// Update the hppMap with the hosts
-	if hppMap != nil {
-		for _, host := range hosts {
-			if !utils.HasElem(hppMap.Host, host) {
-				hppMap.Host = append(hppMap.Host, host)
+	// Update the hosts in the redirect policy of parent VS
+	for _, policy := range o.HttpPolicyRefs {
+		for j := range policy.RedirectPorts {
+			uniqueHosts := make(map[string]struct{})
```

Comment:
@swathinsankaran : Should we use `set` or any other in-built DS that provide uniqueness Instead of handling this in our code?

Reply:
I have changed the code to use the sets library from `k8s.io` pkg.